rotkehlchen/chain/ethereum/modules/curve/crvusd/decoder.py

```diff
@@ -0,0 +1,170 @@
+from rotkehlchen.chain.ethereum.utils import should_update_protocol_cache, token_normalized_value
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.curve.constants import CPT_CURVE
+from rotkehlchen.chain.evm.decoding.curve.lend.common import CurveBorrowRepayCommonDecoder
+from rotkehlchen.chain.evm.decoding.interfaces import ReloadableDecoderMixin
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.globaldb.cache import globaldb_get_general_cache_values
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import CacheType, ChecksumEvmAddress
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class CurvecrvusdDecoder(CurveBorrowRepayCommonDecoder, ReloadableDecoderMixin):
+    """Decoder for crvUSD market events."""
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',  # pylint: disable=unused-argument
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.crvusd = Asset('eip155:1/erc20:0xf939E0A03FB07F59A73314E73794Be0E57ac1b4E').resolve_to_evm_token()  # noqa: E501
+
+    def reload_data(self) -> Mapping['ChecksumEvmAddress', tuple[Any, ...]] | None:
+        """Check that cache is up to date and refresh cache from db.
+        Returns a fresh addresses to decoders mapping."""
+        if should_update_protocol_cache(CacheType.CURVE_CRVUSD_CONTROLLERS) is True:
+            query_crvusd_controllers(evm_inquirer=self.evm_inquirer)
+        elif len(self.controllers) != 0:
+            return None  # we didn't update the globaldb cache, and we have the data already
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            self.controllers = set(globaldb_get_general_cache_values(  # type: ignore[arg-type]  # strings from cache will be valid addresses
+                cursor=cursor,
+                key_parts=(
+                    CacheType.CURVE_CRVUSD_CONTROLLERS,
+                    str(self.evm_inquirer.chain_id.serialize_for_db()),
+                ),
+            ))
```

Comment:
since you're doing a `set()`, you can easily use set comprehension and use `string_to_evm_address()` -- no need for the type ignore.